x-pack/libbeat/management/fleet/manager.go

```diff
@@ -0,0 +1,289 @@
+	"github.com/elastic/beats/x-pack/libbeat/management/api"
+
+	xmanagement "github.com/elastic/beats/x-pack/libbeat/management"
+)
+
+var errEmptyAccessToken = errors.New("access_token is empty, you must re-enroll your Beat")
+
+// ConfigManager provides a functionality to retrieve config channel
+// using which manager is informed about config changes
+type ConfigManager interface {
+	ConfigChan() chan<- map[string]interface{}
+}
+
+func init() {
+	management.Register("x-pack-fleet", NewFleetManager, feature.Beta)
+}
+
+// Manager handles internal config updates. By retrieving
+// new configs from Kibana and applying them to the Beat
+type Manager struct {
+	config    *Config
+	cache     *xmanagement.Cache
+	logger    *logp.Logger
+	beatUUID  uuid.UUID
+	done      chan struct{}
+	registry  *reload.Registry
+	wg        sync.WaitGroup
+	blacklist *xmanagement.ConfigBlacklist
+
+	configChan chan map[string]interface{}
+}
+
+// NewFleetManager returns a X-Pack Beats Central Management manager
+func NewFleetManager(config *common.Config, registry *reload.Registry, beatUUID uuid.UUID) (management.ConfigManager, error) {
+	c := defaultConfig()
+	if config.Enabled() {
+		if err := config.Unpack(&c); err != nil {
+			return nil, errors.Wrap(err, "parsing central management settings")
+		}
+	}
+	return NewFleetManagerWithConfig(c, registry, beatUUID)
+}
+
+// NewFleetManagerWithConfig returns a X-Pack Beats Central Management manager
+func NewFleetManagerWithConfig(c *Config, registry *reload.Registry, beatUUID uuid.UUID) (management.ConfigManager, error) {
+	var cache *xmanagement.Cache
+	var blacklist *xmanagement.ConfigBlacklist
+
+	if c.Enabled && c.Mode == xmanagement.ModeFleet {
+		var err error
+
+		// Initialize configs blacklist
+		blacklist, err = xmanagement.NewConfigBlacklist(c.Blacklist)
+		if err != nil {
+			return nil, errors.Wrap(err, "wrong settings for configurations blacklist")
+		}
+
+		// Initialize central management settings cache
+		cache = &xmanagement.Cache{}
+		if err := cache.Load(); err != nil {
+			return nil, errors.Wrap(err, "reading central management internal cache")
+		}
+	}
+
+	log := logp.NewLogger(management.DebugK)
+
+	return &Manager{
+		config:     c,
+		cache:      cache,
+		blacklist:  blacklist,
+		logger:     log,
+		done:       make(chan struct{}),
+		beatUUID:   beatUUID,
+		registry:   registry,
+		configChan: make(chan map[string]interface{}),
+	}, nil
+}
+
+// Enabled returns true if config management is enabled
+func (cm *Manager) Enabled() bool {
+	return cm.config.Enabled && cm.config.Mode == xmanagement.ModeFleet
+}
+
+// ConfigChan returns a channel used to communicate configuration changes.
+func (cm *Manager) ConfigChan() chan<- map[string]interface{} {
+	return cm.configChan
+}
+
+// Start the config manager
+func (cm *Manager) Start() {
+	if !cm.Enabled() {
+		return
+	}
+
+	cfgwarn.Beta("Central management is enabled")
+	cm.logger.Info("Starting central management service")
```

Comment:
Let's make sure we use a scoped logger that mention that  we are using fleet as the configuration source. We can do this by using `log.Named("fleet")`, this will returned a scoped logger.